ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/integration/PauseResumeIntegrationTest.java

```diff
@@ -0,0 +1,279 @@
+
+  @BeforeClass
+  public static void classSetUp() throws IOException {
+    File BACKUP_LOCATION = TMP_FOLDER.newFolder();
+
+    REST_APP = TestKsqlRestApp
+        .builder(TEST_HARNESS::kafkaBootstrapServers)
+        .withProperty(KSQL_METASTORE_BACKUP_LOCATION, BACKUP_LOCATION.getPath())
+        .withProperty(SESSION_TIMEOUT_MS_CONFIG, 10000) // Reduces the time for a rebalance during ksqlDB restart.
+        .build();
+  }
+
+  @ClassRule
+  public static final RuleChain CHAIN = RuleChain
+      .outerRule(Retry.of(3, ZooKeeperClientException.class, 3, TimeUnit.SECONDS))
+      .around(TEST_HARNESS);
+
+  @Before
+  public void setUpTopicsAndStreams() {
+    REST_APP.start();
+    int CURRENT_ID = COUNTER.getAndIncrement();
+
+    String pageViewPrefix = "PAGEVIEW" + CURRENT_ID;
+    PAGE_VIEWS_PROVIDER = new PageViewDataProvider(pageViewPrefix);
+    PAGE_VIEWS_PROVIDER2 = new PageViewDataProvider(
+        pageViewPrefix, Batch.BATCH2);
+    PAGE_VIEW_TOPIC = PAGE_VIEWS_PROVIDER.topicName();
+    PAGE_VIEW_STREAM = PAGE_VIEWS_PROVIDER.sourceName();
+    SINK_TOPIC = "sink_topic" + CURRENT_ID;
+    SINK_STREAM = "sink_stream" + CURRENT_ID;
+
+    TEST_HARNESS.ensureTopics(PAGE_VIEW_TOPIC);
+    RestIntegrationTestUtil.createStream(REST_APP, PAGE_VIEWS_PROVIDER);
+  }
+
+  @After
+  public void cleanUp() {
+    REST_APP.closePersistentQueries();
+    REST_APP.dropSourcesExcept();
+    REST_APP.stop();
+  }
+
+  @Test
+  public void shouldPauseAndResumeQuery() {
+    // Given:
+    createQuery("");
+    TEST_HARNESS.produceRows(PAGE_VIEW_TOPIC, PAGE_VIEWS_PROVIDER, KAFKA, JSON, System::currentTimeMillis);
+
+    Supplier<Integer> supplier = () -> RestIntegrationTestUtil.makeQueryRequest(REST_APP,
+        "select * from " + SINK_STREAM + ";",Optional.empty()).size();
+
+    // 7 records are produced + a header & footer.
+    assertThatEventually(supplier, equalTo(9));
+
+    // When:
+    // JNH: factor out?
+    String queryId = ((Queries) RestIntegrationTestUtil.makeKsqlRequest(REST_APP, "SHOW "
+        + "QUERIES;").get(0)).getQueries().get(0).getId().toString();
+    RestIntegrationTestUtil.makeKsqlRequest(REST_APP, "PAUSE " + queryId + ";");
+
+    assertThat(getPausedCount(), equalTo(1));
+
+    // Produce more records
+    TEST_HARNESS.produceRows(PAGE_VIEW_TOPIC, PAGE_VIEWS_PROVIDER2, KAFKA, JSON,
+        System::currentTimeMillis);
+
+    // Observe no new records -- This kinda *doesn't work*.  Hard to prove a negative sometimes...
+    assertThatEventually(supplier, equalTo(9));
+
+    // Then:
+    RestIntegrationTestUtil.makeKsqlRequest(REST_APP, "RESUME "
+        + queryId + ";");
+
+    // 5 more records have been produced
+    assertThatEventually(supplier, equalTo(14));
+    assertThat(getRunningCount(), equalTo(1));
+  }
+
+  @Test
+  public void shouldPauseAndResumeMultipleQueries() {
+    // Given:
+    createQuery("1");
+    createQuery("2");
```

Comment:
What's the purpose of having the second query in this test?

Reply:
The goal of this test is to show that pausing one query does not impact a separate query.  I've updated the test to reflect that a little bit.